js/main.js

```diff
@@ -14,20 +108,85 @@ var formElement = document.querySelector('form');
 var titleEl = document.querySelector('#title');
 var urlEl = document.querySelector('#url');
 var notesEl = document.querySelector('#notes');
+var newEntryImgEl = document.querySelector('#new-entry-image');
 
-function submit(event) {
+function save(event) {
   event.preventDefault();
 
-  var inputValue = {
-    title: titleEl.value,
-    url: urlEl.value,
-    notes: notesEl.value,
-    ID: data.nextEntryId
-  };
+  if (data.editing !== null) {
+    // console.log('in editing');
+
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].ID.toString() === data.editing) {
+        data.entries[i].title = titleEl.value;
+        data.entries[i].url = urlEl.value;
+        data.entries[i].notes = notesEl.value;
+        // console.log(data.entries[i]);
+        // console.log('index number:', i);
+
+        // entriesListEl.children[i].find('img').setAttribute('src', urlEl.value);
+        // entriesListEl.children[i].childNodes[i].childNodes[i].setAttribute('src', urlEl.valiue);
+        entriesListEl.children[i].childNodes[0].childNodes[0].setAttribute('src', urlEl.value);
+        entriesListEl.children[i].childNodes[1].childNodes[0].textContent = titleEl.value;
+        entriesListEl.children[i].childNodes[1].childNodes[1].textContent = notesEl.value;
```

Comment:
Instead of changing the content of the existing DOM tree for this entry, I would generate a new DOM tree with the edited data and use the `replaceWith` method to swap the old DOM tree for the new DOM tree.